zebra-scan/src/service/tests.rs

```diff
@@ -254,3 +256,70 @@ pub async fn scan_service_get_results_for_key_correctly() -> Result<()> {
 
     Ok(())
 }
+
+/// Tests that the scan service registers keys correctly.
+#[tokio::test]
+pub async fn scan_service_registers_keys_correctly() -> Result<()> {
+    for network in Network::iter() {
```

Comment:
I like this change, but I think we should avoid adding the `strum`/`strum_macros` and add our own method to `Network`:

```rust
    /// Returns an iterator with an item for each variant of `Network`
    // TODO: Use default values of `Testnet` variant when adding fields for #7845
    pub fn iter_defaults() -> impl Iterator<Item = Self> {
        [Self::Mainnet, Self::Testnet].into_iter()
    }
```

Reply:
OK, I'll use a manual implementation of the iterator. However, I remember I wanted to keep a manual implementation of something else, and Teor said it was better to have it as a dependency because it was easier to review a dependency bump than to maintain a bit of code. I don't remember what it was, though. 

The burden here will be that we'll need to keep the implementation of the iterator in sync with the struct fields when everyone has forgotten about it or there are other devs. We have other spots like that in Zebra, and we occasionally run into issues.

It's not a big deal either way.

Reply:
Done in [6a49f16](https://github.com/ZcashFoundation/zebra/pull/8281/commits/6a49f16c2efc720b1088dbab5cd2e7e26b3ec698).